### DISCUSSION TITLE
Dynamic Power Limiter ohne Batterie - Nulleinspeisung zu viel ins Netz

### DISCUSSION THREAD

**Discussion by zapfs** [hoylabs/OpenDTU-OnBattery#667] [Q&A] [answered]
Ich habe einen Hoymiles 1500 mit 3 Panelen a 405W.
Ich möchte eine "Nulleinspeisung" mit max. 60 W ins Netz realisieren.
Leider speist der WR bei Überschuß zu viel ein. Das gesetzte Limit ist meiner Meinung nach zu hoch berechnet.

Beispiel (Log aus Konsole):
PowerMeterClass: TotalPower: 789.62
09:05:03.969 > [DPL::loop] ******************* ENTER **********************
09:05:04.060 > [DPL::loop] battery interface disabled, SoC: 0 %, StartTH: 80 %, StopTH: 20 %, SoC age: 1564307 s
09:05:04.179 > [DPL::loop] dcVoltage: 36.30 V, loadCorrectedVoltage: 36.32 V, StartTH: 1.00 V, StopTH: 0.00 V
09:05:04.365 > [DPL::loop] StartTH reached: yes, StopTH reached: no, inverter is producing
09:05:04.469 > [DPL::loop] SolarPT disabled, Drain Strategy: 0, canUseDirectSolarPower: no
09:05:04.574 > [DPL::loop] battery discharging allowed, PowerMeter: 790 W, target consumption: -60 W
09:05:04.677 > [DPL::setNewPowerLimit] 4 channels total, 3 producing channels, scaling power limit
09:05:04.781 > [DPL::setNewPowerLimit] requested: 865 W, (re-)sending limit: 1153 W
09:05:04.880 > [DPL::loop] ******************* Leaving PL, calculated limit: 865 W, requested limit: 1153 W (updated from calculated)
09:05:04.983 > TX ActivePowerControl Channel: 40 --> 51 81 85 60 99 80 13 80 32 81 0B 00 2D 0A 00 00 0E 28 06 
09:05:05.090 > [1564307.434] DPL: waiting for a power limit command to complete

Power aus Netz = 790 W
Max. Einspeisung = 60 W
somit dürfte das Limit 790W + 60W = 850 W sein.

gesendet an den WR wird aber 1153 W

Kann mir das jemand erklären?

Meine Settings:
 
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/77608818/83885e10-414c-4fd9-a0dd-62340eee548f)

Viele Grüße
Stefan

**Comment by spcqike** [comment]
Hallo,

ich ich kann versuchen es zu erklären 😄 

die openDTU-onBattery hat im Hintergrund einen Skalierungsmechanismus. Woher der genau kommt, kann ich nicht sagen. Er ist scheinbar recht umstritten, ist aber so noch in der Software.

die Hoymiles limitieren nicht einfach auf der AC Seite, sondern drosseln die MPPT gleichmäßig. 
Wenn du bspw. mit einem HM-800 nur 200W erzeugen willst, werden beide MPPT (und damit beide EIngänge) gleichmäßig auf 100W gedrosselt. hast du nun nur 1 Paneel angeschlossen, kämen, aufgrund der Hoymiles internen Logik, dennoch nur 100W heraus. 
Um das zu umgehen, prüft die openDTU-onBattery, wie viele Eingänge tatsächlich belegt sind und skaliert entsprechend hoch. für mein hiesiges Beipsiel heißt dass, dass aus den 200W (und damit 100W pro EIngang) einfach 400W (und damit 200W pro EIngang) gemacht werden. Am Ende, hat man einen belegten EIngang und produziert, mit dem übertragenen Limit von 400W, seine gewünschten 200W.

Das gleiche passiert auch bei dir. im Log steht, und du sagst es ja auch, dass nur 3 Eingänge belegt sind.

Nun verhält sich der HM-1500 aber etwas anders, als der HM-800. der 1500 hat 4 Eingänge, wovon sich je 2 einen MPPT teilen.

Ich kann dir nicht genau sagen, ob du auch an einem einzigen Eingang bis zu 750W (also das Limit eines MPPT) produzieren könntest, oder ob trotzdem jeder Eingang auf 325W begrenzt ist.

Frage an dich:
Welche Leistung speist der HM-1500 denn ein, wenn das Limit auf 1153W gestellt wird? Wenn der Wechselrichter intern dennoch jeden Eingang separat drosselt, sollte das Ergebnis ja bei ~865W liegen. 

Wenn er jedoch wirklich den MPPT limitiert, könnte bei dir Eingang1&2 jeweils ~288W erzeugen und Eingang3 auf volle Leistung bis zu 325W erzeugen. (bei 1153W Gesamtlimit halbiert es sich ja pro MPPT, also 576,5W pro MPPT. MPPT1 kann dies auch, mit 288W pro Eingang erreichen, MPPT2 hingegen könnte bei 325W, also dem Maximum eines einzelnen Eingangs, limitieren. Es wäre mal interessant zu erfahren, wie der Hoymiles da in Wirklichkeit reagiert 😄 )

**Reply by zapfs** [reply]
Ok Danke schon mal - ich mach mal ein paar Auswertungen sobald ich mehr produziere, als verbrauche. Diese stelle ich dann mal hier ein.

**Reply by hex2647** [reply]
Das was du hier beschreibst wäre aus meiner Sicht das was ich mir wünschen würde... Ich habe 4 Module verbaut an einem HMS1600... 2-3 davon sind meistens stark verschattet uns eins bekommt volle breitseite Sonne. Das eine wird aber auf 150W (600W/4) beschränkt und dadurch komme ich nur auf 150+(3*20W)=210W... Jemand eine Idee wie ich das "beheben" kann?

**Comment by schlimmchen** [comment]
@spcqike hat das Problem schon auf den Kopf getroffen. Umstritten ist diese Skalierung insbesondere bei mir :grin: Hast du denn tatsächlich nur drei Module angeschlossen? Ist ein Eingang unbenutzt? Und welcher Hoymiles genau ist es? HM-1500? HMT-1500?

Falls du wirklich genau drei Eingänge belegt hast, dann skaliert der DPL um 4/3 (auf 1150W), aber der Inverter teilt das dann vermutlich hälftig auf die MPPTs (deiner hat vermutlich zwei). Dann sind zwar zwei Panele korrekt auf 575W begrenzt, aber das dritte darf auch alleine 575W erzeugen, obwohl es nur 287W dazuleisten sollte. Alles was es mehr leistet, ist zuviel. Völlig korrekt.

Diese Skalierung ist total dämlich. Egal ob oder wer sich angegriffen fühlt, das macht einfach keinen Sinn so. Dann muss man wenigstens die Skalierung korrekt implementieren, unter den Annahmen, die gemacht werden. Aber hier passt das Ergebnis der Skalierung ja noch nicht einmal, wenn die Annahmen (Limit wird zu gleichen Teilen auf die MPPTs verteilt) tatsächlich korrekt waren.

**Reply by zapfs** [reply]
Es handelt sich um einen HM-1500. Ich habe nur drei Module angeschlossen. Dabei sind zwei mit gleicher Ausrichtung und Neigungswinkel an einer Seite des Wechselrichters (1. MPPT) und das Dritte an der anderen Seite, da ich es aus Beschattungsgründen senkrecht an der Wand montieren mußte (2. MPPT).

Ich gebe Dir recht - die Skalierung müßte einstellbar sein. Schließlich möchte ich ja die Einspeisung ins Netzt limitieren. Mein Shelly 3EM zeichnete im Januar bis zu 300W Einspeisung auf, obwohl ich maximal 60W eingestellt habe.

Nun warte ich auf Sonnenschein, um die Werte einmal zeitgleich aufzeichnen zu können.

**Comment by zapfs** [comment]
So jetzt hatte ich mal etwas Sonne (wie auf Bestellung).
Hier ein Beispiel:

Max.  Limit = -60 W
Tatsächlich = -213 W
Differenz = 153 W

Errechnetes Limit, welches an den WR gesendet wurde = 603 W

MPPT 1 = 253,64 + 287,979 = 541,619 W
MPPT 2 = 193,008 W
Summe MPPT's = 734,627 W (AC) --- ergibt 645,6 W auf der DC-Seite

Unten seht ihr im Screenshot die Vertteilung auf die 3 Panele.

Da fehlt mir jetzt der Durchblick warum??? 

Protokoll:
14:37:39.837 > [1670668.517] DPL: waiting for sufficiently recent power meter reading
14:37:39.958 > PowerMeterClass: TotalPower: -212.62
14:37:40.071 > [DPL::loop] ******************* ENTER **********************
14:37:40.180 > [DPL::loop] battery interface disabled, SoC: 0 %, StartTH: 80 %, StopTH: 20 %, SoC age: 1670670 s
14:37:40.402 > [DPL::loop] dcVoltage: 41.30 V, loadCorrectedVoltage: 41.91 V, StartTH: 1.00 V, StopTH: 0.00 V
14:37:40.582 > [DPL::loop] StartTH reached: yes, StopTH reached: no, inverter is producing
14:37:40.784 > [DPL::loop] SolarPT disabled, Drain Strategy: 0, canUseDirectSolarPower: no
14:37:40.983 > [DPL::loop] battery discharging allowed, PowerMeter: -213 W, target consumption: -60 W
14:37:41.088 > [DPL::setNewPowerLimit] 4 channels total, 3 producing channels, scaling power limit
14:37:41.198 > [DPL::setNewPowerLimit] requested: 452 W, (re-)sending limit: 603 W
14:37:41.312 > [DPL::loop] ******************* Leaving PL, calculated limit: 452 W, requested limit: 603 W (updated from calculated)
14:37:41.494 > TX ActivePowerControl Channel: 40 --> 51 81 85 60 99 80 13 80 32 81 0B 00 17 8E 00 00 FF 64 05 
14:37:41.634 > [1670670.282] DPL: waiting for a power limit command to complete

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/77608818/caa5f0c2-8011-4a16-aac6-3bff06ad768a)

Die Panels Nr 1 + 2 sind auf einem MMPT.
Panel Nr. 3 auf dem zweiten.

**Reply by spcqike** [reply]
Irgendwie hast du deine Paneele (im Kopf?) falsch und auch die falschen Zahlen raus geschrieben :) 

> MPPT 1 = 253,64 + 287,979

Das sind die bisher auf den beiden Modulen „geernteten“ gesamten Kilowattstunden. Das ist nicht die aktuelle Leistung. 

Dazu kommt dass das Modul mit 287.979 dein benannten „Nr.1“ alleine an einem Eingang hängt. Das siehst du gut daran, dass es über 300W erzeugt, wohingegen die beiden anderen weit weniger haben. 
dszu kommt: die beiden anderen haben exakt die gleiche Spannung, was bei Parallelschaltung an einem MPPT halt so ist. 

deine Module liefern also auf MPPT1 156.6W + 166.3W = 322.9W 
Auf MPPT2 321.2W

in Summe also 644.1W DC seitig. 

Daraus macht der wechselrichter 613W AC seitig, hält also seine 95% Effizienz. 

Damit liefern grade beide MPPT exakt das gleiche (im Bereich der messunsicherheit) 

Wie du richtig sagst, sendet der DPL ein zu hohes Limit. 
du bräuchtest ~450W. Da nur 3 Eingänge belegt sind, greift die Skalierung. Der DPL sagt „3 von 4 sind belegt also muss um 33% erhöht werden“ und er kommt auf das Limit von ~600W. 

Das ist in dem Fall natürlich falsch, darf doch so jeder MPPT 300W (AC, unter Berücksichtigung der Effizienz gönnt er sich dafür ~315W DC) erzeugen, was sie ja auch tun. 

leider fällt mir auch keine einfache und clevere Möglichkeit ein, wie all die wenns und Abers mit diesen wechselrichtern (mehr Eingänge als MPPT) bei nicht voller Belegung abgefangen werden könnten. 

Nicht mal ein Y Adapter, um das eine Paneel an beide Eingänge eines MPPTs zu betreiben, würde dir dauerhaft und immer helfen. 
Das einfache abschalten der Skalierung ebensowenig. 
Im Gegenteil, wenn du mehr als geschätzt 650W benötigen würdest, würdest du weniger produzieren als gewünscht.


Was noch interessant wäre: Liefert das einzelne Modul auch mal mehr als ~325W? Kann auf einem Eingang bspw 400W DC verarbeitet werden? 
je nach Paneelen wird es wahrscheinlich schwierig das überhaupt mal zu erreichen und zu sehen, in unseren Breitengraden 😄

**Comment by schlimmchen** [comment, marked as answer]
> Kann auf einem Eingang bspw 400W DC verarbeitet werden? je nach Paneelen wird es wahrscheinlich schwierig das überhaupt mal zu erreichen und zu sehen, in unseren Breitengraden 😄

Naja, mit bifazialen Modulen auf entsprechendem Untergrund sind 500W kein Problem, die erreiche ich an sonnigen Tagen eher mühelos.

> leider fällt mir auch keine einfache und clevere Möglichkeit ein, wie all die wenns und Abers mit diesen wechselrichtern (mehr Eingänge als MPPT) bei nicht voller Belegung abgefangen werden könnten.

Ich fürchte, um das richtig zu machen, muss man dann das Wissen darüber, wie die Hoymiles sich verhalten, in Software gießen. Dann kann man fragen: Ich sehe hier einen Inverter aus dieser Serie (das weiß OpenDTU bereits) und der hat also n MPPTs. Wenn jetzt nur ein Modul dran hängt, dann weiß die Software was zu tun ist, wenn denn die Annahme mit der Gleichverteilung des Limits wirklich zutrifft. Wenn alle Eingänge belegt sind, gibt es nichts zu skalieren. Bei dreien kann man es bei diesem Beispiel auch noch bestimmen -- zumindest mit der weiteren Annahme, dass die Module das Limit ausnutzen, sonst wird es wieder nichts. Und was ist bei zweien? Hängen die an einem MPPT oder an zwei? Wie stellen wir das fest? Dann muss in der Datenbank auch noch festgehalten sein, welcher Channel an welchem MPPT hängt?! Also wenn man das Skalieren ernst meint, dann geht es nur so ordentlich.

Und das alles nur, um ein Problem des Inverters zu kaschieren... Der Hoymiles sollte das einfach besser machen bzw. sollten wir User damit leben.

**Reply by spcqike** [reply]
> Wie stellen wir das fest?

Haben die Eingänge nicht feste Nummern? Immerhin kann man jeden Eingang sowohl auslesen (siehe live view) als auch konfigurieren (Name, modulleistung). 

Spätestens, wenn man die Spannung zweier Eingänge vergleicht, sollte es „klar“ werden. Der MPPT hat auf beiden Eingängen die gleiche Spannung. Dass zwei MPPTs exakt die gleiche Spannung haben ist möglich, aber unwahrscheinlich, oder? Zumindest sehe ich es bei meinem HM-600 mit 2 Paneelen nie. 

Aber ja. Da alle Eventualitäten in Software abzufangen, so dass es immer ne punktlandung wird, wird äußerst schwer. 

Dann lieber etwas mehr einspeisen und verschenken, als sich selbst limitieren und Strom einkaufen, obwohl man ihn selbst hätte produzieren können.

**Reply by zapfs** [reply]
Danke für die ausführliche Erklärung. Stimmt - ich habe mich beim Ablesen glatt vertan (Sorry) - aber Du hast es ja aussortiert. 
Die Antwort löst zwarmein Problem nicht, erklärt es aber ausreichend.

Vielleicht hat ja mal jemand einen Geistesblitz um das Thema zu lösen.
